### PR TITLE
ref #681: redirect after success register fix

### PR DESCRIPTION
**Pull request by kzorluoglu**
| Q             | A
| ------------- | ---
| Branch        | 7.1.x
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#681
| License       | MIT

### CODE REVIEW

File: src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php [hunk 1]
```diff
@@ -221,7 +221,8 @@ public function Register($sSuccessURL = null, $sFailureURL = null)
             }
 
             if (true === $bDataValid) {
-                $bDataValid = $oUser->Register();
+                $registrationResult = $oUser->Register();
```
Comment (UlrichKu) [inline src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php:224]: The return type of \TDataExtranetUser::Register() should also be fixed then.

Comment (kzorluoglu) [reply src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php:224]: I fixed first this method, but later restored it again. I thing, this change/fix can produce an error in several places.

Comment (UlrichKu) [reply src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php:224]: Why? It is annotated as bool but in 99% of the cases it returns string.

Comment (kzorluoglu) [reply src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php]: i talked with @fabianbarth-esono about the refactoring for the Method Register, but we both thinked, that should be in many place error generate. But, alright, i have the method refactored and committed.

Comment (UlrichKu) [reply src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php]: The initial code was ok.

I meant only the the return *type* annotation - not the implementation of the method.

Comment (kzorluoglu) [reply src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php:224]: Alright, Annotation updated and initial code is again here.

File: src/ExtranetBundle/objects/db/TDataExtranetUser.class.php [hunk 2]
```diff
@@ -225,7 +225,7 @@ public function Save($bForceUserConfirm = false)
      * @param bool $bForceUserConfirmMail
      * @param bool $bAutoLoginAfterRegistration
      *
-     * @return bool
+     * @return bool|string
```
Comment (UlrichKu) [inline src/ExtranetBundle/objects/db/TDataExtranetUser.class.php:228]: + "the id or false for an error" (?)

Comment (kzorluoglu) [reply src/ExtranetBundle/objects/db/TDataExtranetUser.class.php]: done

**Pull request merged**